src/DotLiquid/Hash.cs

```diff
@@ -48,7 +48,9 @@ private static void FromAnonymousObject(object anonymousObject, Hash hash, bool
 
         private static Action<object, Hash> GetObjToDictionaryMapper(Type type, bool includeBaseClassProperties)
         {
-            if (!mapperCache.TryGetValue(type, out Action<object, Hash> mapper))
+            var cacheKey = type.FullName + (includeBaseClassProperties ? "WithBaseProperties" : "WithoutBaseProperties");

```

Comment:
to use same class with/without base properties at the same time, caching should consider including base properties.